mvlearn/embed/dcca.py

```diff
@@ -0,0 +1,349 @@
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+from .base import BaseEmbed
+import torch
+from ..utils.utils import *
+
+
+class DCCA(BaseEmbed):
+    r"""
+    An implementation of Deep Canonical Correlation Analysis [#1DCCA]_ with
+    PyTorch. It computes projections into a common subspace in order to
+    maximize the correlation between pairwise projections into the subspace
+    from two views of data.
+
+    Parameters
+    ----------
+    input_size1 : int (positive)
+        The dimensionality of the input vectors in view 1.
+    input_size2 : int (positive)
+        The dimensionality of the input vectors in view 2.
+    outdim_size : int (positive), default=2
+        The output dimensionality of the correlated projections. The deep
+        network wil transform the data to this size. If not specified, will
+        be set to 2.
+    layer_sizes1 : list of ints, default=None
+        The sizes of the layers of the deep network applied to view 1 before
+        CCA. For example, if the input dimensionality is 256, and there is one
+        hidden layer with 1024 units and the output dimensionality is 100
+        before applying CCA, layer_sizes1=[1024, 100]. If ``None``, set to
+        [1000, ``self.outdim_size``].
+    layer_sizes2 : list of ints, default=None
+        The sizes of the layers of the deep network applied to view 2 before
+        CCA. Does not need to have the same hidden layer architecture as
+        layer_sizes1, but the final dimensionality must be the same. If
+        ``None``, set to [1000, ``self.outdim_size``].
+    use_all_singular_values : boolean (default=False)
+        Whether or not to use all the singular values in the CCA computation
+        to calculate the loss. If False, only the top outdim_size singular
+        values are used.
+    device : string, default='cpu'
+        The torch device for processing.
+    epoch_num : int (positive)
+        The max number of epochs to train the deep networks.
+    batch_size : int (positive)
+        Batch size for training the deep networks.
+    learning_rate : float (positive), default=1e-3
+        Learning rate for training the deep networks.
+    reg_par : float (positive), default=1e-5
+        Weight decay parameter used in the RMSprop optimizer.
+    print_train_log_info : boolean, default=False
+        Whether or not to print the logging info (training loss at each epoch)
+        when calling DCCA.fit().
+    threshold : float, default=1e-2
+        Threshold difference between successive iteration losses to define
+        convergence and stop training.
+
+    Attributes
+    ----------
+    input_size1 : int (positive)
```

Comment:
in sklearn and the previous semester, I thought we were adding an underscore either before or after attribute name?